Dockerfile.windows

```diff
@@ -250,18 +256,89 @@ RUN `
   `
   Write-Host INFO: Configuring git core.autocrlf...; `
-  C:\git\cmd\git config --global core.autocrlf true; `
+  C:\git\cmd\git config --global core.autocrlf true;
+
+# Ensure all directories exist that we will require below....
+WORKDIR ${GOPATH}\${DOCKER_SRC_REL_LOCATION}\${BUNDLES_REL_LOCATION}
+
+WORKDIR ${GOPATH}
+
+RUN `
+  Function Save-FileIfNonZero() { `
+    Param( `
+      [Parameter(Mandatory=$True, `
+       ParameterSetName="""LogName""")] `
+       $LogName `
+    ) `
+    if (Test-Path $LogName) `
+    { `
+      if ((Get-Item $LogName).length -gt 0) { `
+        $SrcPath=Join-Path -Path $Env:GOPATH -ChildPath $Env:DOCKER_SRC_REL_LOCATION; `
+        $BundlesPath=Join-Path -Path $SrcPath -ChildPath $Env:BUNDLES_REL_LOCATION; `
+        $BundlesMiscPath=Join-Path -Path $BundlesPath -ChildPath 'misc'; `
+        Move-Item -Path $LogName -Destination $BundlesMiscPath; `
+      } else { `
+        Remove-Item $LogName; `
+      } `
+    } `
+  } `
+  Function Build-GoTestSum() { `
+    $GotestsumPath=Join-Path -Path $Env:GOPATH -ChildPath $Env:GOTESTSUM_LOCATION; `
+    $GotestsumBinPath=Join-Path -Path $GotestsumPath -ChildPath "gotestsum.exe"; `
+    if (Test-Path $GotestsumBinPath) { `
+      Write-Host "INFO: gotestsum already available in $cwGotestsumBinPath - skipping building it...." `
+      return `
+    } else { `
+      Write-Host "INFO: No existing binary of gotestsum found. Building gotestsum version $Env:GOTESTSUM_COMMIT in $Env:GOPATH" `
+    } `
+    $optsForGet = @('"get"', '"-d"', '"gotest.tools/gotestsum"'); ` 
+    $logFileName = 'gotestsum.get.out';  `
+    &go $optsForGet > $logFileName 2>&1; `
+    $savedExitCode = $LASTEXITCODE; `
+    Save-FileIfNonZero -LogName $logFileName; `
+    if ($savedExitCode -ne 0) {  `
+      Throw '"Failed getting gotestsum sources..."'  `
+    }; `
+    Write-Host "INFO:     Sources obtained for gotestsum..."; `
+    Push-Location $GotestsumPath; `
+    $optsForCheckout = @('"checkout"', '"-q"', """$Env:GOTESTSUM_COMMIT"""); ` 
+    $logFileName = 'gotestsum.git.out';  `
+    &git $optsForCheckout > $logFileName 2>&1; `
+    $savedExitCode = $LASTEXITCODE; `
+    Save-FileIfNonZero -LogName $logFileName; `
+    if ($savedExitCode -eq 0) { `
+      Write-Host "INFO:     Checkout done for gotestsum..."; `
+      $logFileName = 'gotestsum.build.out';  `
+      $optsForBuild = @('"build"', '"-buildmode=exe"'); ` 
+      &go $optsForBuild > gotestsum.build.out 2>&1; `
+      $savedExitCode = $LASTEXITCODE; `
+      Save-FileIfNonZero -LogName $logFileName; `
+    } else { `
+      Throw '"gotestsum checkout failed..."'; `
+    } `
+    Pop-Location; `
+    `
+    if ($savedExitCode -ne 0) {  `
+      Throw '"gotestsum build failed..."'; `
+    } `
+    Write-Host "INFO:     Build done for gotestsum..."; `
+  } `
   `
-  Write-Host INFO: Completed
+  Write-Host INFO: Building gotestsum...; `
```

Comment:
Should we move this inside the `Build-GoTestSum` function?

Reply:
I do not know a compelling argument either way. Why would you choose one over the other?

Reply:
Done